services/galley/src/Galley/Data/TeamFeatures.hs

```diff
@@ -72,8 +66,7 @@ setFeatureStatusNoConfig ::
   (TeamFeatureStatus a) ->
   m (TeamFeatureStatus a)
 setFeatureStatusNoConfig tid status = do
-  let flag = Public.tfwoStatus status
-  retry x5 $ write (update (Public.knownTeamFeatureName @a)) (params Quorum (flag, tid))
+  retry x5 $ write (update (Public.knownTeamFeatureName @a)) (params Quorum (featureStatus status, tid))
```

Comment:
Can we now get rid of `FeatureHasNoConfig` altogether?  At least in this function and the getter, I think.

Reply:
I'm not so convinced this is an improvement.  I thought there was some simplification in there, but in all cases in this commit we just replace a call to a more specialized function by one to a more general function.  And even if we can drop the `HasNoConfig` constraint here, I'm not sure that's a good thing.  We certainly don't need the more general form now.

I'll take another look at this tomorrow and try to remember why I was suggesting this.  Perhaps it was just a mistake?

Reply:
> I'm not so convinced this is an improvement. I thought there was some simplification in there, but in all cases in this commit we just replace a call to a more specialized function by one to a more general function. And even if we can drop the `HasNoConfig` constraint here, I'm not sure that's a good thing. We certainly don't need the more general form now.
> 
> I'll take another look at this tomorrow and try to remember why I was suggesting this. Perhaps it was just a mistake?

I checked again, and it looks like there is not much to win here.  I tried to simplify things with `DuplicateRecordFields`, and decided I like the extension, but it doesn't work very well [in our case](https://github.com/wireapp/wire-server/blob/smatting/applock-followup-2/services/galley/src/Galley/Data/TeamFeatures.hs#L69):

```
/home/mf/src/wire/wire-server/services/galley/src/Galley/Data/TeamFeatures.hs:69:78: error:
     Ambiguous occurrence ‘featureStatus’
```

I pushed it anyway and reverted it, for your amusement.

I'll leave it up to you whether to keep 6416c63 or revert it, I have a slight preference for removing it (sorry :)).

Reply:
I like the aesthetics better, but yes, seems superfluous. Reverted